src/abstractdataframe/reshape.jl

```diff
@@ -399,3 +399,40 @@ function CategoricalArrays.CategoricalArray(v::RepeatedVector)
     res.refs = repeat(res.refs, inner = [v.inner], outer = [v.outer])
     res
 end
+
+
+"""
+    transpose(df::AbstractDataFrame, src_namescol=1, dest_namescol=names(df)[src_namescol]; copycols::Bool=false, makeunique=false, promote_type::Bool=true)
+
+Transpose a `DataFrame`, such that rows become columns,
```

Comment:
technically what we do is `permutedims`, but probably it is better to call it `transpose` as you propose. Just please make a note in the docstring that `transpose` is not recursive (as opposed to `transpose` in Base).